### PR TITLE
Add the "rules" element in RenameFilterPlugin

### PR DESCRIPTION
**Pull request by dmikurube**
Add the "rules" element in RenameFilterPlugin with initial renaming operators "lower_case_to_upper" and "upper_case_to_lower".

### CODE REVIEW

**Comment by dmikurube** [comment]
@uu59 Can you please take a look? I'm especially wondering if ConfigException is good or not here.

**Comment by uu59** [comment]
Correctly used 👍

File: embulk-standards/src/main/java/org/embulk/standards/RenameFilterPlugin.java [hunk 1]
```diff
@@ -53,4 +70,38 @@ public PageOutput open(TaskSource taskSource, Schema inputSchema,
...
+            }
+        } else {
+            throw new ConfigException("Non-string renaming operator is not supported");
+        }
+    }
+
+    private Schema applyUpperCaseToLower(Schema inputSchema) {
+        Schema.Builder builder = Schema.builder();
+        for (Column column : inputSchema.getColumns()) {
+            builder.add(column.getName().toLowerCase(), column.getType());
```
Comment (uu59) [inline embulk-standards/src/main/java/org/embulk/standards/RenameFilterPlugin.java]: type: timestamp column has a `format`. Probably it should be kept after rename filter through.

Comment (uu59) [inline embulk-standards/src/main/java/org/embulk/standards/RenameFilterPlugin.java]: @muga let me correct if I'm wrong

Comment (dmikurube) [inline embulk-standards/src/main/java/org/embulk/standards/RenameFilterPlugin.java]: If so, existing "columns" should be fixed as well?

**Comment by dmikurube** [comment]
@uu59 Chained rules need to be tested, but the combination of upper_case_to_lower and lower_case_to_upper may not work to test chaining. (Only the latter is meaningful!)

Will add that test with next operators.

File: embulk-standards/src/main/java/org/embulk/standards/RenameFilterPlugin.java (hunk 1 above)
Comment (uu59) [inline embulk-standards/src/main/java/org/embulk/standards/RenameFilterPlugin.java]: @dmikurube Ah sorry you are right. Please ignore my comment

**Comment by dmikurube** [comment]
@uu59 Updated the PR per our discussion to accept map-formed rules.

btw, when or how do we usually decide to merge a PR in Embulk?

**Comment by uu59** [comment]
@dmikurube well, I think one of Embulk maintainers (especially @muga or @frsyuki?) approval then merge this by themselves normally.

**Comment by dmikurube** [comment]
Got it. Thanks!

**Comment by dmikurube** [comment]
@muga @frsyuki I'm just wondering if I could merge this base PR at first, and then add more renaming rules in separate PRs. What do you think?

**Comment by dmikurube** [comment]
Updated the PR per f2f discussion with @muga.

**Comment by dmikurube** [comment]
@frsyuki  
Not to block this base PR due to the naming discussion, I've split the concrete renaming rules  ("convert_...") into another branch for the time being. https://github.com/dmikurube/embulk/tree/renamefilterplugin-case

What do you think of this base PR #483?

**Comment by frsyuki** [comment]
I think that `columns` should run after `rules` because `columns` have more flexible control than `rules`.  Looks good excepting that 👍

**Pull request merged**